### PR TITLE
common/redis: replace library/redis with base-images/redis

### PR DESCRIPTION
**Pull request by majewsky**
Also, simplify some templating to remove useless variability.

### CODE REVIEW

File: common/redis/templates/deployment.yaml [hunk 1]
```diff
@@ -31,8 +36,8 @@ spec:
     spec:
       containers:
       - name: {{ template "redis.fullname" . }}
-        image: {{ template "redis.image" . }}
-        imagePullPolicy: {{ default "" .Values.image.pullPolicy | quote }}
+        image: {{ $registry }}/base-images/redis:{{ .Values.image.tag }}
+        imagePullPolicy: {{ if contains "latest" .Values.image.tag }}Always{{ else }}IfNotPresent{{ end }}
```
Comment (SuperSandro2000) [inline common/redis/templates/deployment.yaml:40]: That's smart, I like it.

File: common/redis/templates/_helpers.tpl [hunk 2]
```diff
@@ -12,31 +12,6 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | replace "_" "-" | trimSuffix "-" -}}
```
Comment (SuperSandro2000) [inline common/redis/templates/_helpers.tpl:12]: We should combine those two functions tbh

File: common/redis/templates/_helpers.tpl [hunk 3]
```diff
@@ -12,31 +12,6 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
 
-{{/*
-  The (contains "." $repo) checks if the chart user has overridden the
-  repository to refer to a custom registry. If not, the default Docker Hub
-  mirror gets used.
-*/}}
-{{- define "redis.image" -}}
-  {{- $repo := required ".Values.image.repository missing" .Values.image.repository -}}
-  {{- $tag := required ".Values.image.tag missing" .Values.image.tag -}}
-  {{- if contains "." $repo -}}
```
Comment (SuperSandro2000) [inline common/redis/templates/_helpers.tpl:23]: What? If the repo contains a dot, we are just not using the mirror!?...

**Review by SuperSandro2000 — APPROVED** [review]

File: common/redis/templates/_helpers.tpl (hunk 3 above)
Comment (majewsky) [reply common/redis/templates/_helpers.tpl:23]: That's for image refs that already contain a hostname, e.g. Keppel forcing the alternate region mirrors.

**Pull request merged**